Create seed with coin flips

BtcAutoNode opened the pull request:
## Description

_Describe the change simply. Provide a reason for the change._
Creation of seeds via entropy input from coin flips.

_Include screenshots of any new or modified screens (or at least explain why they were omitted)_
![grafik](https://github.com/user-attachments/assets/90fc761b-9920-48e7-84d3-5ff2200a5c2e)
![grafik](https://github.com/user-attachments/assets/2626b54d-27b4-4a5f-939d-f6e222e49876)



This pull request is categorized as a:

- [X] New feature
- [ ] Bug fix
- [ ] Code refactor
- [ ] Documentation
- [ ] Other

## Checklist

- [ ] I’ve run `pytest` and made sure all unit tests pass before sumbitting the PR


If you modified or added functionality/workflow, did you add new unit tests?

- [ ] No, I’m a fool
- [x] Yes
- [ ] N/A

![grafik](https://github.com/user-attachments/assets/198889b1-0da8-45d7-8c59-e8245989c990)

I have tested this PR on the following platforms/os:

- [ ] Raspberry Pi OS [Manual Build](https://github.com/SeedSigner/seedsigner/blob/dev/docs/manual_installation.md)
- [ ] [SeedSigner OS](https://github.com/SeedSigner/seedsigner-os) on a Pi0/Pi0W board
- [X] SeedSigner Emulator
- [ ] Other


Note: Keep your changes limited in scope; if you uncover other issues or improvements along the way, ideally submit those as a separate PR. The more complicated the PR the harder to review, test, and merge.

BtcAutoNode committed:
Update tools_views.py

- add menu entry and views

BtcAutoNode committed:
Update components.py

- add icons (coins ion for the seed menu and T/H for head and tails)

BtcAutoNode committed:
Update tools_screens.py

- add coin flip screen

BtcAutoNode committed:
Update mnemonic_generation.py

- add number of flips constants

BtcAutoNode committed:
Update tools_screens.py

- change layout to 3, 3 similar to the dice screen

BtcAutoNode committed:
Update tools_screens.py

- change order of H and T

BtcAutoNode committed:
Update tools_views.py

- add missing space in menu entry

BtcAutoNode committed:
Update test_mnemonic_generation.py

- add pytest tests similar to the tests for dice rolls

BtcAutoNode committed:
Update test_mnemonic_generation.py

- fix copy/paste error

kdmukai commented on src/seedsigner/views/tools_views.py:
```diff
@@ -236,6 +240,58 @@ def run(self):
+class ToolsCoinEntropyMnemonicLengthView(View):
+    def run(self):
+        TWELVE = f"12 words ({mnemonic_generation.COIN__NUM_FLIPS__12WORD} flips)"
+        TWENTY_FOUR = f"24 words ({mnemonic_generation.COIN__NUM_FLIPS__24WORD} flips)"
```
Moving these constants to be class-level attrs will allow us to specify 12 vs 24 in various FlowTest scenarios.

kdmukai commented:
I think this was my fault (definitely not yours), but now that we're actually exposing the coin flip code to users, `generate_mnemonic_from_coin_flips` should:

* Verify the length of the coin flip input string.
* Verify that the contents are only "0" or "1".

I realize that the `KeyboardScreen.return_after_n_chars` gives us the same length guarantees, but it's better for the `mnemonic_generation` helper to be bulletproof.

kdmukai commented:
Confirmed that the unit tests match the expected results in iancoleman.io

kdmukai commented:
Would ideally have 2-3 FlowTests:

* Create 12-word
* Create 24-word
* Start creating, but exit early

(the current `test_seed_flows.py` has lots of missing FlowTests for existing functionality, unfortunately)

kdmukai commented:
Also, tagging in @easyuxd for thoughts on the "H" and "T" keyboard icons. They're being pulled from the FontAwesome pack, not our UI font.

kdmukai commented:
Should also include new screenshots in `tests/screenshot_generator/generator.py`

easyuxd commented:
Since we're trying to get away from using FontAwesome icons, can we swap the "H" and "T" for the regular Open Sans characters?

Adding a coinflip icon to our SeedSigner icon set is a to-do for me, so I'm okay with using this FontAwesome stack-of-coins icon as a placeholder in the menu until I can replace it.

BtcAutoNode commented:
@easyuxd : Would this be better? 1)
![grafik](https://github.com/user-attachments/assets/67988c65-bd37-4817-8453-1a49e232ba16)
Or this one: 2)
![grafik](https://github.com/user-attachments/assets/95cebb26-a27b-43ef-92b9-97afbd7b4209)

Initially I tried to do it exactly like the dice roll seed creation. Only copying the ...dice... functions/constants etc and changed 'dice' to 'coin'. I had a hard time finding icons and decided to use H and T, but so, using these awesomefont characters the buttons had the same size as in the dice creation screen.
This is also a matter of what someone likes more or not (the 2 screens I've added here). 

Or without the additional texts: 3)
![grafik](https://github.com/user-attachments/assets/27cb5fbb-f62e-4343-b8f9-a6e5ec2da781)
And: 4)
![grafik](https://github.com/user-attachments/assets/87e9fb2f-aef3-4ede-9edb-afde0e9b01f0)

@kdmukai : As mentioned in the chat, the same changes as proposed above will need to be done for the dice roll scenario as well, I guess. 
I was only able to do this with my 0% python experience if I can copy and paste existing code and check what needs to be changed (like we only have 2 values, different icons etc). So without these changes in the dice creation I have nothing to look at and re-use ,)

BtcAutoNode commented:
I've just recognized by chance that in PR534 someone already added coding for the coin flip seed creation, hm. 
And in the Seedsigner dev branch in the tools_screens.py file there is already a function for the coin flip (same as in this PR I think). So I am a little confused right now.
This one here:
![grafik](https://github.com/user-attachments/assets/6ea3171f-f94e-41b3-82b6-540dbd65b0fb)

easyuxd commented:
@BtcAutoNode 
Much better with the change in font, thanks! I prefer the full-width layout of #2. It's more balanced, and the counter at the bottom is useful and it helps understand the H and T labels.

I can't speak to a previous coin-flip effort, either @kdmukai or @newtonick should know.

BtcAutoNode committed:
Update components.py

- delete heads/tails icon (and stick to normal letters instead of icons (as easyuxd proposed)

BtcAutoNode committed:
Update tools_screens.py

- changed coin flip screen) to reflect changes proposed by easyuxd)

ayush3504 commented:
Hello Everyone!
I would like to work on this PR as a part of SOB. I wanted to know if I should open a pull request in this PR branch itself or should I open  a new PR separately?

BtcAutoNode commented:
> Hello Everyone! I would like to work on this PR as a part of SOB. I wanted to know if I should open a pull request in this PR branch itself or should I open a new PR separately?

If you could take it over or create a new one from it, it would be best.

kdmukai commented:
Thanks for the getting this PR started @BtcAutoNode and for handing it off. It sounds like it would make sense to close this PR.
